apps/methodologies/bold/rule-processors/offer/rewards-distribution/src/lib/rewards-distribution.helpers.ts

```diff
@@ -67,127 +66,98 @@ export const calculateAmount = (options: {
-      participants.set(key, {
-        ...value,
-        actorTypes: assert<Participant['actorTypes']>(
-          arrayOfUniqueValues([
-            ...value.actorTypes,
-            DocumentEventActorType.REMAINDER,
-          ]),
-        ),
-        amount: remainder.amount.plus(value.amount).toString(),
-        percentage: remainder.percentage.plus(value.percentage).toString(),
+  const { actors, remainder } = options;
+
+  for (const [key, actor] of actors) {
+    if (actor.actorType === DocumentEventActorType.NETWORK) {
+      actors.set(key, {
+        ...actor,
+        amount: remainder.amount.plus(actor.amount).toString(),
+        percentage: remainder.percentage.plus(actor.percentage).toString(),
       });
 
       break;
     }
   }
 };
 
-export const addParticipantsPercentage = (options: {
-  massTotalValue: BigNumber;
-  participants: Map<string, Participant>;
-  unitPrice: NonZeroPositive;
-}): void => {
-  const { massTotalValue, participants, unitPrice } = options;
-
-  const amountTotal = formatDecimalPlaces(
-    massTotalValue.multipliedBy(unitPrice),
-  );
-
-  for (const [key, value] of participants) {
-    const amount = new BigNumber(value.amount);
-
-    options.participants.set(key, {
-      ...value,
-      percentage: calculateOfferPercentage(amount, amountTotal),
-    });
-  }
-};
+const calculateMassTotalValue = (massValues: BigNumber[]) =>
+  sumBigNumbers(massValues);
 
 export const aggregateCertificateRewards = (
   unitPrice: NonZeroPositive,
   resultContentsWithMassValue: ResultContentWithMassValue[],
 ): AggregateCertificateRewards => {
-  const participants = new Map<string, Participant>();
+  const actors: ActorsByActorType = new Map();
+
+  const massTotalValue = calculateMassTotalValue(
+    resultContentsWithMassValue.map(({ massValue }) => massValue),
+  );
 
-  let massTotalValue = new BigNumber(0);
+  const offerTotal = formatDecimalPlaces(
+    massTotalValue.multipliedBy(unitPrice),
+  );
 
   for (const {
     massValue,
     resultContent: { certificateRewards },
   } of resultContentsWithMassValue) {
-    massTotalValue = massTotalValue.plus(massValue);
-
     for (const certificateReward of certificateRewards) {
-      const {
-        actorTypes,
-        certificatePercentage,
-        participant: { id, name },
-      } = certificateReward;
-      const participant = participants.get(id);
-
-      participants.set(id, {
-        actorTypes: assert<Participant['actorTypes']>(
-          arrayOfUniqueValues([
-            ...actorTypes,
-            ...(participant?.actorTypes ?? []),
-          ]),
-        ),
-        amount: calculateAmount({
-          certificatePercentage,
-          massValue,
-          participantAmount: participant?.amount,
-          unitPrice,
-        }),
-        name,
-        percentage: '',
+      const { actorType, participant, percentage } = certificateReward;
+      const actor = actors.get(`${actorType}-${participant.id}`);
```

Comment:
pra reduzir código duplicado e evitar possíveis inconsistências, uma alternativa seria criar uma constante pro actor key
```suggestion
      const actorKey = `${actorType}-${participant.id}`;
      const actor = actors.get(actorKey);
```